test/nullable-validation.test.js

```diff
@@ -44,15 +43,16 @@ test('nullable string', t => {
     method: 'POST',
     url: '/',
     body: {
-      hello: null
+      hello: undefined
```

Comment:
Why has this changed?

Reply:
.same matches both maybe? That's what I assumed but would like to get a clarification

Reply:
By putting `null`, the test failed.

This is the code fragment i used:

```js
....

  fastify.inject({
    method: ‘POST’,
    url: ‘/’,
    body: {
      hello: null
    }
  }, (err, res) => {
    t.assert.ifError(err)
    t.assert.deepStrictEqual(res.payload.hello, null)
    done()
  })

....
```
returning this error ( see screenshot ).

<img width="757" alt="screenshot-test" src="https://github.com/user-attachments/assets/0b9d1d1b-1b6e-4510-bc3f-387772f68e34" />

I tried many ways but to no avail so I put `undefined` in place of `null`, if you have other ideas however let me know so that we make other changes if necessary. 🙂

Reply:
The `undefined` value (an invalid JSON value) is being stripped by the request, while this test wants to ensure that the `null` value (a valid JSON value) is propagated from the request till the response.

So we must keep the `null` here.

TBH I think `res.payload` is a string and `.hello` returns `undefined`.
Try `res.json().hello` instead.

Reply:
ok change made, see commit here:  [6bcb7ba](https://github.com/fastify/fastify/pull/5880/commits/6bcb7ba882369b14385134765b8ae6ca674baca4)